tests/testthat/test-social-smells.R

```diff
@@ -0,0 +1,112 @@
+test_that("Using Perceval version other than 0.12.24 will fail for parse_mbox", {
+  perceval_path <- "../../../tools/bin/perceval-incompatible-version"
+  mbox_path <- "../../../../rawdata/mbox/project.mbox"
+  expect_error(parse_mbox(perceval_path, mbox_path))
+})
+test_that("Correct config file parameters and Perceval version 0.12.24 work", {
+  perceval_path <- "../../../tools/bin/perceval"
+  mbox_path <- "../../../../rawdata/mbox/project.mbox"
+  jira_issue_comments_path <- "../../../../rawdata/issue_tracker/project_issue_comments.json"
+  expect_no_error(parse_mbox(perceval_path, mbox_path))
+  expect_no_error(parse_jira(jira_issue_comments_path))
+})
+test_that("Incorrect paths to files passed to parse functions fails", {
```

Comment:
Hi Carlos, I'm a little confused on this. 

If we were testing if "parse_mbox fails when the path to perceval binary is incorrect", wouldn't we want to make a hardcoded path to perceval that we know is incorrect instead of taking the user's path from tools.yml, which may or may not be correct?  

For testing something like "parse_mbox fails when the path to mbox binary is incorrect", I understand why we would take the user's tools.yml path to perceval. However, shouldn't the mbox path be manually set by us to be a bad path instead of reading it from a project config file?

Reply:
> If we were testing if "parse_mbox fails when the path to perceval binary is incorrect", wouldn't we want to make a hardcoded path to perceval that we know is incorrect instead of taking the user's path from tools.yml, which may or may not be correct?

I think I got confused about what you were trying to test here, the parse_mbox() function, or the fact the user typed it wrong. If you are testing the function behavior to a wrong path, then you are correct you should hardcode.